### ISSUE TITLE
bug with WATCHgetWFDarealPrecip function

### ISSUE THREAD

**Issue by earthcli** [CentreForHydrology/Reanalysis#2]
I wonder if there is abug in the WATCHgetWFDarealPrecip, about the subset of all.values by [land_selected$Land,]

selected_values <- all.values[land_selected$Land,]

nrow of all.values was 67420 as length of land$Land,
while land$Land has a value range of 8925-209746, so 
there should be the rowid inputed to the all.values, not the land_selected$Land?

length(unique(land$Land))
[1] 67420
range(unique(land$Land))
[1]   8925 209746

all.values[209746,]
# error!! : incorrect number of dimensions

**Comment by earthcli** [comment]
and I also wonder if it is safe to bind land$Land and all.values with no subset,
like result =data.frame(land$Land, all.values), I means if it is in the right order?
I have checked that glon is sorted, while the glat is unsorted in the land dataframe.

**Comment by KevinShook** [comment]
Thanks - I'll look into it.

**Comment by KevinShook** [comment]
You were quite right about the error - thanks very much!
I'm now using which() to find the locations. I did this in the other functions, but must have forgotten in thsi one (I've never actually used it!).
I'll push it to github tonight.

**Comment by earthcli** [comment]
Thanks very much for your response and also for the excellent work on the Reanalysis package.
I have sucessuffly read my data with reference to your script, but I have a question, 
where is your land dataset from? I did not find it in the web of: http://www.eu-watch.org.
I wonder the glat and latitude column in the land data.frame is not matched.
with the function glat (71.25) result in a glat of 323, which is not matched(38) with the land data.frame (first row). while glat (-71.25) result in a glat of 38
so, is the glat column also downloaded from the web set?

  Land Longitude Latitude Ht.m. glon  glat   
 26641   -179.75    71.25   170      1      38

I wonder if I can create the land data.frame by myselt, with the dimmesion information of the nc file,
for example, first create a matrix of 720 rows and 360 cols, and then create a landid start with 1,end with 259200,  finally,the Latitude and Longitude vector can be create as follow:
watchlat <-  function(glat){
  seq(-89.75,89.75,0.5)[glat]
}

watchlon <-  function(glon){
  seq(-179.75,179.75,0.5)[glon]
}
but this result in a different data.frame vs the land data.frame from your package, would you give me some advice?

Thanks again!

**Comment by KevinShook** [comment]
Thanks very much for your comments and for finding the bug - I really appreciate it.
The values are from a file called WFD-land-long-lat-z.dat. It was among the files that were downloaded with the WATCH data. The data was downloaded by one of our grad students, so I'm not sure where it came from.
This is the first line
   Land Longitude Latitude Ht(m)  glon glat
   26641  -179.75   71.25   170.0    1   38  
I've attached the file, but had to change the extension for GitHub to accept it.
[WFD-land-long-lat-z.txt](https://github.com/CentreForHydrology/Reanalysis/files/1516193/WFD-land-long-lat-z.txt)

The functions glat and glon aren't the same as the columns in the file - I just used the same names.


K
